jps-plugin/src/org/intellij/erlang/jps/builder/ErlangDependencyTreeNode.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2012-2015 Sergey Ignatov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.intellij.erlang.jps.builder;
+
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.xmlb.annotations.AbstractCollection;
+import com.intellij.util.xmlb.annotations.Attribute;
+import com.intellij.util.xmlb.annotations.Tag;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class ErlangDependencyTreeNode {
+  @Attribute("path")
+  public String path;
+
+  @Tag("moduleInfo")
+  public SourceRootInfo mySourceRootInfo;
+
+  @Tag("dependencies")
+  @AbstractCollection(surroundWithTag = false, elementTag = "dependency")
+  public List<String> myDependencies = ContainerUtil.newArrayList();
+
+  public ErlangDependencyTreeNode(@NotNull String path,
+                                  @Nullable SourceRootInfo sourceRootInfo,
+                                  @NotNull List<String> dependencies) {
+    this.path = path;
+    this.mySourceRootInfo = sourceRootInfo;
+    myDependencies = dependencies;
+  }
+
+  @SuppressWarnings("unused") // reflection
+  public ErlangDependencyTreeNode() {
+  }
+
+  public ErlangDependencyTreeNode(String filePath, List<String> dependencies) {
+    this(filePath, null, dependencies);
+  }
+
+  public static class SourceRootInfo {
```

Comment:
So you've decided to store module name for each file path. What's the reason for that? You can compute this information in your builder.